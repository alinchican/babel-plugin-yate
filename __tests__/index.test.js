const babel = require("@babel/core");
const plugin = require("../");

test("should return original string if translation is not found", () => {
  const testCode = "customTagName`Welcome user to ${place}!`";

  const { code } = babel.transform(testCode, {
    plugins: [
      [
        plugin,
        {
          tagName: "customTagName",
          translationsFile: "./__mocks__/translations.js",
        },
      ],
    ],
  });
  expect(code).toMatchSnapshot();
});

test("should throw when translation file is not found", () => {
  const testCode = "t`Hello to ${place}, ${name}!`";

  expect(() =>
    babel.transform(testCode, {
      plugins: [[plugin]],
    })
  ).toThrow();
});

test("should transform code with custom tagName", () => {
  const testCode = "customTagName`Welcome to ${place}, ${name}!`";

  const { code } = babel.transform(testCode, {
    plugins: [
      [
        plugin,
        {
          tagName: "customTagName",
          translationsFile: "./__mocks__/translations.js",
        },
      ],
    ],
  });

  expect(code).toMatchSnapshot();
});

test("should transform code with default tagName", () => {
  const testCode = "t`Welcome to ${place}, ${name}!`";

  const { code } = babel.transform(testCode, {
    plugins: [
      [
        plugin,
        {
          translationsFile: "./__mocks__/translations.js",
        },
      ],
    ],
  });

  expect(code).toMatchSnapshot();
});

test("should skip code not matching tagName", () => {
  const testCode = "random`Welcome to ${place}, ${name}!`";

  const { code } = babel.transform(testCode, {
    plugins: [
      [
        plugin,
        {
          translationsFile: "./__mocks__/translations.js",
        },
      ],
    ],
  });

  expect(code).toMatchSnapshot();
});

test("should throw when using multiple string context variables", () => {
  const testCode = "t('default', 'test')`Hello to ${place}, ${name}!`";

  expect(() =>
    babel.transform(testCode, {
      plugins: [[plugin]],
    })
  ).toThrow();
});

test("should throw with invalid context variable type (array)", () => {
  const testCode = "t(['default'])`Hello to ${place}, ${name}!`";

  expect(() =>
    babel.transform(testCode, {
      plugins: [[plugin]],
    })
  ).toThrow();
});

test("should return translation using function without context", () => {
  const testCode = "t()`Welcome to ${place}, ${name}!`";

  const { code } = babel.transform(testCode, {
    plugins: [
      [
        plugin,
        {
          translationsFile: "./__mocks__/translations.js",
        },
      ],
    ],
  });

  expect(code).toMatchSnapshot();
});

test("should return translation using function with default context", () => {
  const testCode = "t('default')`Welcome to ${place}, ${name}!`";

  const { code } = babel.transform(testCode, {
    plugins: [
      [
        plugin,
        {
          translationsFile: "./__mocks__/translations.js",
        },
      ],
    ],
  });

  expect(code).toMatchSnapshot();
});

test("should return translation containing repeated parameters", () => {
  const testCode =
    "t`Offer available from ${minAmount} ${currency} to ${maxAmount} ${currency}`";

  const { code } = babel.transform(testCode, {
    plugins: [
      [
        plugin,
        {
          translationsFile: "./__mocks__/translations.js",
        },
      ],
    ],
  });

  expect(code).toMatchSnapshot();
});

test("should return translation containing strings looking like parameters", () => {
  const testCode = "t`Hello ${`Bucharest`}`";

  const { code } = babel.transform(testCode, {
    plugins: [
      [
        plugin,
        {
          translationsFile: "./__mocks__/translations.js",
        },
      ],
    ],
  });

  expect(code).toMatchSnapshot();
});
