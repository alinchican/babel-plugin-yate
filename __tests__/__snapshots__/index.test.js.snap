// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return original string if translation is not found 1`] = `"\`Welcome user to \${place}!\`;"`;

exports[`should return translation containing repeated parameters 1`] = `"\`Offer available from \${minAmount} \${currency} to \${maxAmount} \${currency}\`;"`;

exports[`should return translation containing strings looking like parameters 1`] = `"\`Salutare {Budapest}, pardon, \${\`Bucharest\`}\`;"`;

exports[`should return translation using function with default context 1`] = `"\`Salut \${name}, bine ai venit la \${place}!\`;"`;

exports[`should return translation using function without context 1`] = `"\`Salut \${name}, bine ai venit la \${place}!\`;"`;

exports[`should skip code not matching tagName 1`] = `"random\`Welcome to \${place}, \${name}!\`;"`;

exports[`should transform code with custom tagName 1`] = `"\`Salut \${name}, bine ai venit la \${place}!\`;"`;

exports[`should transform code with default tagName 1`] = `"\`Salut \${name}, bine ai venit la \${place}!\`;"`;
